tst/utils/ThreadsafeBlockingQueue.cpp

```diff
@@ -132,12 +132,12 @@ void* readingThread(void* ptr) {
 TEST_F(ThreadsafeBlockingQueueFunctionalityTest, multithreadQueueDequeueTest)
 {
     PSafeBlockingQueue pSafeQueue = NULL;
-    SafeQueueUser user;
+    SafeQueueUser user = {0};
     UINT64 totalThreads = 0, item = 0, threadCount = 0;
     BOOL empty = FALSE;
     TID threads[8] = {0};
-    srand(GETTIME());
-    totalThreads = rand()%7 + 2;
+    SRAND((UINT32) GETTIME());
```

Comment:
Why type cast to UIN32? I see this on linux: `void srand (unsigned int seed);` where unsigned int could be 32 / 64 bit, correct? Wondering, what issue this was causing on Windows?

Reply:
It's an unfortunate side effect of C portability. Int started off as the machine native word but later in 32 bit days it got adopted as 32 bits and is the standard on Windows 32/64 and Unix. The warnings was generated casting 64 bit return of GETTIME to 32 bits which is a truncation. Windows compilers have better detection for many of these types of warnings.

Reply:
Regarding the tests... There were actually a number of tests that were failing when run locally. Some were due to the tests being bad. On Windows, the unitialized param will be that - an unitialized. On Linux platforms, most of the time it gets zeroed. There were other tests that were flaky due to timing as the OSes can chose to have longer quantization of the scheduler which improves the performance greatly for many service-type scenarios. Thus, on my local Windows platform, my scheduler is 30ms as opposed to 5ms which is what many other OS-es are configured as out of the box (Windows Data Center gets configured with 100ms I believe). Yet, there were also bugs regarding the int getting cast into boolean...?